Frameworks/CoreGraphics/CGContext.mm

```
@@ -362,6 +362,20 @@ inline HRESULT GetTextRenderingParams(IDWriteRenderingParams* originalParams, ID
         return S_OK;
     }
 
+    inline HRESULT Flush() {
+        // We cannot use use ID2D1::Flush here, mainly due to:
+        // 1. Flush flushes the D2D batch, but does not flush the D3D batch, we do not have a way to flush
+        // the D3D batch for WicBitmapRenderTarget. (Not: During the EndDraw procedure this occurs.)
+        // 2. Sometimes a show copy of our buffer can be used to do the actual rendering, Flush does not force the copy
+        // over of the shadow copy into the original buffer (EndDraw does)
+        // Thus the best way is to implement Flush is through EndDraw
+        if (_beginEndDrawDepth > 0) {
+            RETURN_IF_FAILED(deviceContext->EndDraw());
```

This will discard the error value. We have to go through PopEndDraw to collect all of the error values from all of the draw operations.